src/ai/contexts.cpp

```diff
@@ -981,9 +981,9 @@ const std::set<map_location>& keeps_cache::get()
 			for(int y = 0; y != map_->h(); ++y) {
 				const map_location loc(x,y);
 				if(map_->is_keep(loc)) {
-					map_location adj[6];
-					get_adjacent_tiles(loc,adj);
-					for(size_t n = 0; n != 6; ++n) {
+					adjacent_loc_array_t adj;
+					get_adjacent_tiles(loc,adj.data());
+					for(size_t n = 0; n < adj.size(); ++n) {
```

Comment:
size_t is not necessarily the same as unsigned int, which you're using elsewhere.

Not sure it's worth changing. More likely we should grep for size_t and reconsider its use everywhere, because it's used far to often instead of unsigned int, which would be more appropriate in most cases.

Reply:
But `std::array::size` returns `size_t`, so we should use `size_t` here.